apps/api/src/donations/helpers/donation-status-updates.ts

```diff
@@ -1,21 +1,50 @@
 import { DonationStatus } from '@prisma/client'
 
+const initial: DonationStatus[] = [DonationStatus.initial]
+const changeable: DonationStatus[] = [
+  DonationStatus.incomplete,
+  DonationStatus.paymentRequested,
+  DonationStatus.waiting,
+]
+const final: DonationStatus[] = [
+  DonationStatus.succeeded,
+  DonationStatus.cancelled,
+  DonationStatus.deleted,
+  DonationStatus.declined,
+  DonationStatus.invalid,
+  DonationStatus.refund,
+]
+
+function isInitial(status: DonationStatus) {
+  return initial.includes(status)
+}
+
+function isChangeable(status: DonationStatus) {
+  return changeable.includes(status)
+}
+
+function isFinal(status: DonationStatus) {
+  return final.includes(status)
+}
 /**
  * The function returns the allowed previous status that can be changed/updated by the incoming donation event
  * @param newStatus the incoming status of the payment event
  * @returns allowed previous status that can be changed by the event
  */
-export function getAllowedPreviousStatus(newStatus: DonationStatus): DonationStatus | undefined {
-  switch (newStatus) {
-    case DonationStatus.waiting: {
-      return DonationStatus.initial
-    }
-    case DonationStatus.succeeded: {
-      return DonationStatus.waiting
-    }
-    case DonationStatus.cancelled: {
-      return DonationStatus.waiting
-    }
+export function shouldAllowStatusChange(
+  oldStatus: DonationStatus,
+  newStatus: DonationStatus,
+): boolean {
+  if (isFinal(oldStatus) || isInitial(newStatus)) {
+    return false
+  }
```

Comment:
updating when old === new should also be possible

Reply:
Check [add a check on shouldAllowStatusChange to return true when the st…](https://github.com/podkrepi-bg/api/pull/436/commits/73ad3b4c3295c3932053380f1d58d37d5038ca28)